### ISSUE TITLE
`dev.ts` breaks node compat for mongodb

### ISSUE THREAD

**Issue by Kyiro** [denoland/fresh#1374]
Using a newly generated Fresh project, adding the code snippet below to `main.ts` and running the project via `dev.ts` breaks mongodb for whatever reason. Starting with `main.ts` works perfectly fine.
```ts
import { MongoClient } from "npm:mongodb";

const mongoClient = new MongoClient("mongodb://localhost:27017");

await mongoClient.connect();
```
```ts
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'stderr')
    at resolveLogPath (file:///C:/Users/Kyiro/AppData/Local/deno/npm/registry.npmjs.org/mongodb/5.6.0/lib/mongo_logger.js:110:38)
    at Function.resolveOptions (file:///C:/Users/Kyiro/AppData/Local/deno/npm/registry.npmjs.org/mongodb/5.6.0/lib/mongo_logger.js:343:29)
    at parseOptions (file:///C:/Users/Kyiro/AppData/Local/deno/npm/registry.npmjs.org/mongodb/5.6.0/lib/connection_string.js:397:66)      
    at new MongoClient (file:///C:/Users/Kyiro/AppData/Local/deno/npm/registry.npmjs.org/mongodb/5.6.0/lib/mongo_client.js:48:63)```

**Comment by jiawei397** [comment]
This is a bug in Deno, and there are still issues with support for NPM. I have encountered it before and raised an [issue](https://github.com/denoland/deno/issues/18984), but my sample has been officially resolved. The deeper issues have not been resolved yet. You can consider writing a demo and then raising an issue for them to solve.

My temporary solution is to add a `devStart` method, which can be used like this:
```ts
import { devStart } from "$fresh/dev.ts";

await devStart(import.meta.url);
await import("./main.ts");
```

In other words, the original `dev` method is being split up:
```ts
export async function dev(base: string, entrypoint: string) {
  await devStart(base);
  const entry = new URL(entrypoint, base).href;
  await import(entry);
}
```
This is just an idea. If needed, I can submit a pull request. In the meantime, you can also test it temporarily using my repository at https://deno.land/x/jw_fresh@v0.7.2/.

**Comment by hey-ab** [comment]
My dirty hack

```typescript
// mongo_client.ts
import { MongoClient } from "mongodb";

export const mongoClient = new MongoClient(Deno.env.get("MONGODB_URI")!);
```

```typescript
// dev.ts
#!/usr/bin/env -S deno run -A --watch=static/,routes/

import dev from "$fresh/dev.ts";
import { mongoClient } from "./mongo_client.ts"; // Yes, just import, no idea why

await dev(import.meta.url, "./main.ts");
```
And `await mongoClient.connect()` anywhere in your project

**Comment by Kyiro** [comment]
`stderr` is used on the MongoClient constructor, so if you use it before dynamically importing `main.ts`, it'll work. My hack around it was to completely get rid of `dev.ts` and make `fresh.gen.ts` dynamic but that won't work on deploy or in other projects as I'm mounting fresh on top of Hono.
